src/cobra/test/test_util/test_context.py

```diff
@@ -0,0 +1,42 @@
+"""Test functions of context.py."""
+
+import pytest
+
+from cobra.util import HistoryManager, get_context, resettable
+
+
+def test_history_manager():
+    """Test initialization and resetting of HistoryManager."""
+    # initialize HistoryManager
+    history_manager = HistoryManager()
+    # add non-functioning operation
+    history_manager(lambda: 1)
+    # reset operations
+    history_manager.reset()
+
+
+def test_get_context(model):
+    """Test if context retrieval is working."""
+    with model as model:
+        # reverse optimization direcion
+        model.objective_direction = "min"
+        context = get_context(model)
+        # should have history
+        assert context._history[-1].args[-1] == "max"
+
+    # there shouldn't be any history
+    with pytest.raises(IndexError):
+        context._history[-1]
+
+
+def test_resettable():
+    """Test if resettable decorator is functional."""
+    # decorate a dummy function
+    @resettable
+    def change_my_name(old_name, new_name):
+        """Change old name to new name."""
+        if old_name != new_name:
+            old_name = new_name
+
+    # call the dummy function
+    change_my_name("hmm", "hmmm")
```

Comment:
It seems like some assertion should be made here?

Reply:
I am not sure what assertion to use here. The idea is that the function should not throw any errors during runtime. The way it's implemented requires the method to be already a part of `cobra.core.Model` class (if the test should be related to the `Model`'s functionality). And, so I just made sure it works and thus the test function doesn't return anything useful. Any suggestion though?

Reply:
I don't really. Sorry.

Reply:
Can we let it be for now and revisit later when we have some better idea?